### ISSUE TITLE
Entries field within Matrix block doesn't limit results to current site when eager-loaded

### ISSUE THREAD

**Issue by lukeyouell** [craftcms/cms#5451]
### Description

We've found that when an entry field sits within a matrix block, that selected entries are returned to the templates even though they aren't enabled for the current site, but are enabled for a different one. We've confirmed this behaviour in a fresh installation.

#### Matrix block field settings

Handle: fooblocks
Propagation method: Only save blocks to the site they were created in

#### Site A

  - Matrix block (fooblocks)
    - Entries field (entries)
      - Entry A (Disabled for both sites)
      - Entry B (Enabled for Site A)
      - Entry C (Disabled for Site A but enabled for Site B)
      - Entry D (Enabled for both sites)

```
No eager-loading: {{ entry.fooblocks.first.entries.all|column('title')|join(', ') }}
Returns: Entry B, Entry D <-- correct

With eager-loading: {{ clone(entry.fooblocks).with(['block:entries']).first.entries|column('title')|join(', ') }}
Returns: Entry B, Entry C, Entry D <-- incorrect (Entry C shouldn't be returned)
```

#### Site B

  - Matrix block (fooblocks)
    - Entries field (entries)
      - Entry A (Disabled for both sites)
      - Entry B (Enabled for Site A)
      - Entry C (Disabled for Site A but enabled for Site B)
      - Entry D (Enabled for both sites)

```
No eager-loading: <pre>{{ entry.fooblocks.first.entries.all|column('title')|join(', ') }}</pre>
Returns: Entry C, Entry D <-- correct

With eager-loading: {{ clone(entry.fooblocks).with(['block:entries']).first.entries|column('title')|join(', ') }}
Returns: Entry B, Entry C, Entry D <-- incorrect (Entry B shouldn't be returned)
```

### Additional info

- Craft version: 3.3.20.1 & 3.4-RC1

**Comment by brandonkelly** [comment]
Thanks for reporting that! This was happening due to some left over code from d8e579b8533259cb42ff3aad3785572be81a7080, which was supposed to be removed in 12cbf2ce5af1e11e69ad75ad9c532831799529ab (per your prior issue #4751). Fixed for good now, for the next 3.4 release.

**Comment by lukeyouell** [comment]
@brandonkelly is there an approximate ETA for 3.4? This issue affects a pretty big site of ours that is currently running 3.3. Is there any chance of this fix being applied to 3.3?

Thanks for sorting it!

**Comment by brandonkelly** [comment]
Assuming no major outstanding bugs, it will be released on January 28. And since that’s right around the corner, pretty good chance there won’t be any additional 3.3 releases.
